src/legacy/core_plugins/data/public/index_patterns/index_patterns_service.mock.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexPatternsService, IndexPatternsSetup } from '.';
+import { FieldType } from './fields';
+import { StaticIndexPattern } from './index_patterns';
+// @ts-ignore
+import { flattenHitWrapper } from './index_patterns/flatten_hit';
+
+type IndexPatternsServiceClientContract = PublicMethodsOf<IndexPatternsService>;
+
+const createSetupContractMock = () => {
+  // Legacy mock - must be removed before migrating to new platform.
+  // Included here because we only want to call `jest.mock` when somebody creates
+  // the mock for this contract.
+  jest.mock('ui/chrome');
```

Comment:
Does this mock have to be here? It might be unexpected if user wants to themselves mock `ui/chrome`, but it is already mocked.

Reply:
Yeah I debated whether to mock `ui/chrome` in each of the individual tests that were using index patterns, vs mocking it in one place here.

The reason I settled on this implementation was because I felt we should minimize the amount of stuff people need to know about dependencies of their dependencies. Each of the tests knows it is using index patterns, so it is reasonable to ask them to use `jest.mock('ui/index_patterns')`. And they know they are (eventually) moving over to new platform, so it's reasonable to ask them to `jest.mock('ui/new_platform')`. 

But they don't necessarily know that index patterns have moved to the data plugin, which starts up some legacy stuff, which relies on chrome. It felt unfair to ask them to figure this out, which is why I moved the mock here instead.

There are a few scenarios where this has downsides though:
1. Case you mentioned: Users want to provide custom mocks for `chrome`, or require the "real" `chrome`. In this case they'd need to unmock or `requireActual`.
2. Users writing tests with index patterns have other code that needs `chrome` to be mocked as well. Their tests are "magically" working, but may break sometime in the future when we remove `chrome` (and the mock) from index patterns.

Overall I think scenario 1 will be relatively uncommon for most users. Scenario 2 could be more common, but is still something we can fix in the PR where we remove chrome from index patterns.